src/replica/replica_disk_balancer.cpp

```diff
@@ -0,0 +1,116 @@
+                       req.target_disk,
+                       enum_to_string(status()),
+                       enum_to_string(_disk_replica_migration_status));
+        resp.err = ERR_BUSY;
+        return false;
+    }
+
+    if (status() != partition_status::type::PS_SECONDARY) {
+        derror_replica("received disk replica migration(gpid={}, origin={}, target={}, "
+                       "partition_status={}), but replica must be PS_SECONDARY",
+                       req.pid.to_string(),
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(status()));
+        resp.err = ERR_INVALID_STATE;
+        return false;
+    }
+
+    bool valid_origin_disk = false;
+    bool valid_target_disk = false;
+    for (const auto &dir_node : _stub->_fs_manager._dir_nodes) {
+        if (dir_node->tag == req.origin_disk) {
+            valid_origin_disk = true;
+            std::set<gpid> disk_holding_replicas = dir_node->holding_replicas[req.pid.get_app_id()];
+            if (disk_holding_replicas.find(req.pid) == disk_holding_replicas.end()) {
+                derror_replica("received disk replica migration(gpid={}, origin={}, target={}, "
+                               "partition_status={}), but replica doesn't exist on origin disk",
+                               req.pid.to_string(),
+                               req.origin_disk,
+                               req.target_disk,
+                               enum_to_string(status()));
+                resp.err = ERR_OBJECT_NOT_FOUND;
+                return false;
+            }
+        }
+
+        if (dir_node->tag == req.target_disk) {
+            valid_target_disk = true;
+            std::set<gpid> disk_holding_replicas = dir_node->holding_replicas[req.pid.get_app_id()];
+            if (disk_holding_replicas.find(req.pid) != disk_holding_replicas.end()) {
+                derror_replica("received disk replica migration(gpid={}, origin={}, target={}, "
+                               "partition_status={}), but replica has existed on target disk",
+                               req.pid.to_string(),
+                               req.origin_disk,
+                               req.target_disk,
+                               enum_to_string(status()));
+                resp.err = ERR_PATH_ALREADY_EXIST;
+                return false;
+            }
+        }
+    }
+
+    if (!valid_origin_disk || !valid_target_disk) {
+        derror_replica("received disk replica migration(gpid={}, origin={}, target={}, "
+                       "partition_status={}), but disk is not existed",
+                       req.pid.to_string(),
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(status()));
+        resp.err = ERR_OBJECT_NOT_FOUND;
+        return false;
+    }
+
+    set_disk_replica_migration_status(disk_replica_migration_status::MOVING);
```

Comment:
I think you should move these out of this funciton. because the name of this func is `check_***`, it can't do things out of range of check work.